comps/dataprep/src/integrations/arangodb.py

```diff
@@ -0,0 +1,456 @@
+
+        try:
+            llm_transformer = LLMGraphTransformer(
+                llm=llm,
+                allowed_nodes=allowed_nodes,
+                allowed_relationships=allowed_relationships,
+                prompt=prompt_template,
+                node_properties=node_properties or False,
+                relationship_properties=relationship_properties or False,
+                ignore_tool_usage=ignore_tool_usage,
+            )
+        except (TypeError, ValueError) as e:
+            if logflag:
+                logger.warning(f"Advanced LLMGraphTransformer failed: {e}")
+            # Fall back to basic config
+            try:
+                llm_transformer = LLMGraphTransformer(llm=llm, ignore_tool_usage=ignore_tool_usage)
+            except (TypeError, ValueError) as e:
+                if logflag:
+                    logger.error(f"Failed to initialize LLMGraphTransformer: {e}")
+                raise e
+
+        ########################################
+        # Text Embeddings Inference (optional) #
+        ########################################
+
+        if OPENAI_API_KEY and OPENAI_EMBED_ENABLED:
+            # Use OpenAI embeddings
+            self.embeddings = OpenAIEmbeddings(
+                model=OPENAI_EMBED_MODEL,
+                dimensions=OPENAI_EMBED_DIMENSION,
+            )
+        elif TEI_EMBEDDING_ENDPOINT and HUGGINGFACEHUB_API_TOKEN:
+            # Use TEI endpoint service
+            self.embeddings = HuggingFaceHubEmbeddings(
+                model=TEI_EMBEDDING_ENDPOINT,
+                huggingfacehub_api_token=HUGGINGFACEHUB_API_TOKEN,
+            )
+        elif TEI_EMBED_MODEL:
+            # Use local embedding model
+            self.embeddings = HuggingFaceBgeEmbeddings(model_name=TEI_EMBED_MODEL)
+        else:
+            raise ValueError("No embeddings environment variables are set, cannot generate embeddings.")
+
+        ############
+        # ArangoDB #
+        ############
+
+        client = ArangoClient(hosts=ARANGO_URL)
+        sys_db = client.db(name="_system", username=ARANGO_USERNAME, password=ARANGO_PASSWORD, verify=True)
+
+        if not sys_db.has_database(ARANGO_DB_NAME):
+            sys_db.create_database(ARANGO_DB_NAME)
+
+        db = client.db(name=ARANGO_DB_NAME, username=ARANGO_USERNAME, password=ARANGO_PASSWORD, verify=True)
+        if logflag:
+            logger.info(f"Connected to ArangoDB {db.version()}.")
+
+        self.llm_transformer = LLMGraphTransformer(
+            llm=llm,
+            allowed_nodes=allowed_nodes,
+            allowed_relationships=allowed_relationships,
+            prompt=prompt_template,
+            node_properties=node_properties or False,
+            relationship_properties=relationship_properties or False,
+            ignore_tool_usage=ignore_tool_usage,
+        )
+        self.graph = ArangoGraph(db=db, include_examples=False, generate_schema_on_init=False)
```

Comment:
I had to slightly modify the parameter names for `ArangoGraph`:
```suggestion
        self.graph = ArangoGraph(db=db, generate_schema_on_init=False, schema_include_examples=False)
```